### PR TITLE
Add manifest for dev.storyapps.starc

### PR DESCRIPTION
**Pull request by joshua-stone**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Links:**

* https://discord.com/channels/819491465734389840/819491465734389843/1032264309880930434
* https://discord.com/channels/819491465734389840/983004591425716224/1032267557316599828

- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

CC: @dimkanovikov

### CODE REVIEW

**Comment by joshua-stone** [comment]
bot, build dev.storyapps.starc

**Comment by hfiguiere** [comment]
pr should be against `new-pr` branch.

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: this is disappointing.

File: dev.storyapps.starc.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      - install -Dm755 starc-run                  -t ${FLATPAK_DEST}/bin
+      - mv squashfs-root ${FLATPAK_DEST}/bin/starc
+    sources:
+      - type: script
+        dest-filename: starc-run
+        commands:
+          - /app/bin/starc/starc $@
+      - type: file
+        filename: starc-setup.AppImage
+        url: https://github.com/story-apps/starc/releases/download/v0.3.1/starc-setup.AppImage
```
Comment (hfiguiere) [inline dev.storyapps.starc.yml:36]: why not building from source instead?

File: dev.storyapps.starc.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+app-id: dev.storyapps.starc
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: starc-run
+separate-locales: false
+finish-args:
+  - --socket=x11
```
Comment (hfiguiere) [inline dev.storyapps.starc.yml:8]: missing `--share=ipc`

**Pull request closed without merging**